import axios from "axios";

const LOCAL_API_BASE_URL = "http://localhost:8080";

const RECIPE_SEARCH_BASE_URL = "https://api.edamam.com/api/recipes/v2?type=public";
const RECIPE_SEARCH_APP_KEY = "f7b7c5856e28029b8e7ab08182ab9fe1"
const RECIPE_SEARCH_APP_ID = "9cf8e5c7";

// TODO: Implement the searchMeals function
// export const searchRecipes = async (q) => {
//   try {
//     const response = await axios.get(`${RECIPE_SEARCH_BASE_URL}&app_key=${RECIPE_SEARCH_APP_KEY}&app_id=${RECIPE_SEARCH_APP_ID}&q=${q}`);
//     return response.data;
//   } catch (error) {
//     console.error("There was an error fetching the recipes!", error);
//     throw error;
//   }
// };

export const fetchMeals = async () => {
  try {
    const response = await axios.get(`${LOCAL_API_BASE_URL}/api/meals/saved`, {withCredentials:true});
    console.log("response", response);
    return response.data;
  } catch (error) {
    console.error("There was an error fetching the meals!", error);
    throw error;
  }
};

export const addMeal = async (meal) => {
  try {
    // console.log("uri", uri, "label", label, "image", image, "source", source);
    const response = await axios.post(`${LOCAL_API_BASE_URL}/api/meals/new`, meal,
      {
        headers: { 'content-type': 'application/json' },
        withCredentials: true,
      });
      console.log("response.data", response.data);
      return response.data;
  } catch (error) {
    console.error("There was an error saving the meal!", error);
    throw error;
  }
};


export const deleteRecipe = async (recipeId) => {
  try {
    await axios.post(`${LOCAL_API_BASE_URL}/api/meals/delete`, null, {
      params: { recipeId },
      withCredentials: true,
    });
  } catch (error) {
    console.error("There was an error deleting the meal!", error);
    throw error;
  }
};
